src/app/screens/LNURLPay.tsx

```diff
@@ -106,13 +107,36 @@ function LNURLPay(props: Props) {
         comment, // https://github.com/fiatjaf/lnurl-rfc/blob/luds/12.md
         payerdata, // https://github.com/fiatjaf/lnurl-rfc/blob/luds/18.md
       };
-      const { data: paymentInfo } = await axios.get<LNURLPaymentInfo>(
-        details.callback,
-        {
-          params,
+
+      let response;
+
+      try {
+        response = await axios.get<LNURLPaymentInfo | LNURLPaymentInfoError>(
+          details.callback,
+          {
+            params,
+            // https://github.com/fiatjaf/lnurl-rfc/blob/luds/01.md#http-status-codes-and-content-type
+            validateStatus: () => true,
+          }
+        );
+
+        const isSuccessResponse = function (
```

Comment:
Do you know the reason the spec decided against HTTP status codes? Doesn't that make the implementation / tooling more complicated for all services consuming those APIs? 

If you can think of a more elegant way to solve that, please let me know!

Reply:
nice, implementation. 

No idea why status codes should not be respected. honestly I am still OK with respecting them. If a server for example returns a 500 as a valid response, then imo the server is so broken, that I do not respect the response :) 

I think in other places we do assume a success response. 
but this PR looks small and clean to me.  👍